src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -564,6 +564,42 @@ static bool IsRelevantDocument(SyntaxTreeIndex syntaxTreeInfo)
             }
         }
 
+        protected Task<ImmutableArray<FinderLocation>> FindReferencesInElementAccessExpressionsAsync(
+            ISymbol symbol,
+            Document document,
+            SemanticModel semanticModel,
+            bool isFindMethod,
+            CancellationToken cancellationToken)
+        {
+            return FindReferencesInDocumentAsync(symbol, document, IsRelevantDocument, CollectMatchingReferences, cancellationToken);
+
+            static bool IsRelevantDocument(SyntaxTreeIndex syntaxTreeInfo)
+                => syntaxTreeInfo.ContainsElementAccessExpression;
+
+            void CollectMatchingReferences(ISymbol originalUnreducedSymbolDefinition, SyntaxNode node,
+                ISyntaxFactsService syntaxFacts, ISemanticFactsService semanticFacts, ArrayBuilder<FinderLocation> locations)
+            {
+                ISymbol elementAccessMethodOrProperty;
+                if (isFindMethod)
+                {
+                    elementAccessMethodOrProperty = semanticFacts.GetElementAccessMethod(semanticModel, node);
+                }
+                else
+                {
+                    elementAccessMethodOrProperty = semanticFacts.GetElementAccessCountableProperty(semanticModel, node);
+                }
+
+                if (Matches(elementAccessMethodOrProperty, originalUnreducedSymbolDefinition))
+                {
+                    var location = node.GetFirstToken().GetLocation();
```

Comment:
this locatoin doesn't seem appropriate.  if i have `a.b.c[...]` wouldn't this select the `a`?  can you add appropriate tests.